app/src/main/java/com/android/unio/model/firestore/FirestoreReferenceList.kt

```diff
@@ -72,7 +72,16 @@ class FirestoreReferenceList<T>(
 
   /** Requests all documents from Firestore and updates the list. */
   override fun requestAll(onSuccess: () -> Unit) {
+    // If the list is already up-to-date, return early.
+    val fetchedUids = _list.value.map { it.uid }
+    if (fetchedUids == _uids) {
+      onSuccess()
+      return
+    }
```

Comment:
Very good idea ! However, if you keep this implementation, you won't fetch the database even if other components are modified remotely, given that the list of uids is the same. This is quite critical as it can happen in various context. 

As I am not far from you and I thus asked you this question, you proposed to add a boolean "forcedPush" which can bypass the request if we need to see changes from the database. I agree with you and this is a good way of handling the problem ! However I think we would like to see the updates from Firebase in almost every situation, given that we are in the Screen displaying them. The answer might be in [realtime updates with Cloud Firestore](https://firebase.google.com/docs/firestore/query-data/listen) -> 

We could have a ListenerManager that manage all the listeners to the database according to the current screen. If the current one is the Event list for example, with only 3 events visible on the page, we could say that there is one listener per event visible, and if we want to be meticulous, even only for the specific fields seen in the screen. Then if the user click on one event, we will kill the listeners of the two other events and add fields to the listener of the clicked event. This is just an idea, and of course outside the scope of this PR ; )

Reply:
Adding `Snapshot` listeners for sure requires our attention, I read about it while I was investigating offline mode.
 I don't understand quite well how this ListenerManager would work, but I'd be happy to talk about it tomorrow if you think it's worth it. 

However I've seen some practices of people defining an observer in their repositories, that listens to the database (that would also work locally, e.g. in offline mode) and update its contents (association list, event list,...). So I agree, `Snapshot` listeners are something we'll need to use, I think. We'll discuss it soon anyway.

Reply:
Indeed I agree with you both. For now I have added a `lazy` parameter to `.requestAll` which when set to true exits the function earlier. Its default value is false.